File: python/5.py
## kstrikis' solution for project euler problem 4
## released under The Unlicense
## What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
## the answer will be the product of all primes less than 20 times
## any repeated factors (e.g. 16 = 2*2*2*2 and 9 = 3*3)
## for 1 to 20, these repeated factors are 2, 2, 2, and 3
def isprime(n,i):
    if(n==2):
        return 1
    if(n%i==0):
        return 0
    if(i**2>n):
        return 1
    return isprime(n,i+1)
product = 1
for x in range(20):
        if(isprime(x,2)):
            product = product*x
print(product*8*3)

File: c/5.c
// kstrikis' solution for project euler problem 5
// released under The Unlicense
// What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
// compile with gcc 5.c -lm -o 5
// answer will be product of primes < 20 times repeated factors (16=2*2*2*2, 9=3*3)
#include<stdio.h>
#include<math.h>
#include<stdbool.h>

bool isprime(int n, int i) {
	if (n==2) return true;
	if (i*i>n) return true;
	if (n%i==0) return false;
	return isprime(n,i+1);
}
int main(){
	int prod = 1;
	for(int x=1; x<20; x++) {
		if(isprime(x,2)) prod *= x;
	}
	printf("%d\n",prod*8*3);
}
